tiledb/sm/array_schema/array_schema.cc

```diff
@@ -538,6 +554,62 @@ Status ArraySchema::drop_attribute(const std::string& attr_name) {
   return Status::Ok();
 }
 
+Status ArraySchema::rename_attribute(
+    const std::string& old_name,
+    const std::string& new_name,
+    bool is_new_uri = true) {
+  std::lock_guard<std::mutex> lock(mtx_);
+  if (old_name.empty() || new_name.empty()) {
+    // Do nothing if name is empty
+    return Status::Ok();
```

Comment:
Maybe throw an error in this case instead?

Reply:
Changed to throw an ArraySchemaError